src/stdlib/lib.c

```diff
@@ -203,7 +207,7 @@ value_p to_string(value_p val) {
 			double possible_num = val->numericVal;
 			int rounded_int = (int) lrint(possible_num);
 			char *converted_str;
-			if (fabs(possible_num - rounded_int) < 1e-7) {
+			if (fabs(possible_num - rounded_int) < FLOAT_CUTOFF) {
```

Comment:
Eventually we should consider the float cutoff to be relative to the rounded_int. If the value is 0.00000001, then this method will not work well.